File: 500102181/insertion_sort.c
#include <stdio.h>

void print_array(int *arr, int n)
{
    for (int i = 0; i < n; i++)
        printf("%d\t", arr[i]);
}

void insertion_sort(int *arr, int n)
{
    int key, j;
    for (int i = 1; i <= n - 1; i++)
    {
        key = arr[i];
        j = i - 1;
        while (j >= 0 && arr[j] > key)
        {
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = key;
    }
}

int main()
{
    int n;
    printf("Enter size of array : ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter elements in array : ");
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);
    printf("Unsorted Array : ");
    print_array(arr, n);
    printf("\n==============================================================================\n");
    insertion_sort(arr, n);
    printf("Sorted Array : ");
    print_array(arr, n);
    return 0;
}

File: 500102181/Python/InsertionSort.py
def InsertionSort(list,n):
    for i in range(n-1):
        key = list[i]
        j = i - 1
        while(j>=0 and list[j] > key):
            list[j+1] = list[j]
            j-=1
        list[j+1] = key

list = [2,4,3,9,1,4,8,7,5,6]
n = len(list)
print("Unsorted list...\n")
print(list)
InsertionSort(list,n)
print("Sorted list...\n")
print(list)
